### ISSUE TITLE
Doesn't work on iOS with forms

### ISSUE THREAD

**Issue by hktari** [halkar/xamarin-range-slider#115]
slider not visible on iOS and after suggested code is added:

#if NETFX_CORE
[assembly: Xamarin.Forms.Platform.WinRT.ExportRenderer(, typeof(Xamarin.RangeSlider.Forms.RangeSliderRenderer))]
#else
[assembly: Xamarin.Forms.ExportRenderer(typeof(Xamarin.RangeSlider.Forms.RangeSlider), typeof(Xamarin.RangeSlider.Forms.RangeSliderRenderer))]
#endif

NullReferenceException thrown

project:
[Test.zip](https://github.com/halkar/xamarin-range-slider/files/1618422/Test.zip)

**Comment by halkar** [comment]
Thanks @hktari 
Apparently, range slider ios package is not compatible with new iOS projects. I will try to fix it asap.

**Comment by mwerickson** [comment]
Has there been any resolution to this?  I am using XF 2.5.0.121934 and the slider is not visible on iOS.

**Comment by Magic73** [comment]
I got this exception:

Loaded assembly: /.........../Xamarin.RangeSlider.Common.dll [External]
Unhandled Exception:

System.NullReferenceException: Object reference not set to an instance of an object

But from source code, Xamarin.RangeSlider.Common.dll doesn't contain anything.

I compiled the source code with samples, and it works.... 

the latest nuget version, is the same of the actual source code??

**Comment by Magic73** [comment]
I just noticed that on iOS sample, inside the main.cs you put:

    public class Application
    {
        // This is the main entry point of the application.
        static void Main(string[] args)
        {
            // if you want to use a different Application Delegate class from "AppDelegate"
            // you can specify it here.
            UIApplication.Main(args, null, "AppDelegate");

            **var t = typeof(RangeSliderRenderer);**
        }
    }

**Comment by Magic73** [comment]
So the issue is inside this:

        public override CGSize IntrinsicContentSize => new CGSize(NoIntrinsicMetric,
            Math.Max(LowerHandleImageNormal.Size.Height, UpperHandleImageNormal.Size.Height) + SpaceAboveThumbs);

it seems that LowerHandleImageNormal is null, and LowenHandleImageNormal is:


 private UIImage LowerHandleImageNormal
        {
            get
            {
                if (_lowerHandleImageNormal != null)
                    return _lowerHandleImageNormal;
                if (IS_PRE_IOS7())
                {
                    var image = ImageFromBundle(Enabled ? @"slider-default-handle" : @"slider-default-handle-disabled");
                    _lowerHandleImageNormal = image.ImageWithAlignmentRectInsets(new UIEdgeInsets(0, 2, 0, 2));
                }
                else
                {
                    var image = ImageFromBundle(Enabled ? @"slider-default7-handle" : @"slider-default7-handle-disabled");
                    _lowerHandleImageNormal = image.ImageWithAlignmentRectInsets(new UIEdgeInsets(-1, 8, 1, 8));
                }

                return _lowerHandleImageNormal;
            }
        }

**Comment by Magic73** [comment]
Ok.. now I just copied all the slider*.png into my Resorce folder of the iOS project, and now it works.

halkar, may you fix it?

**Comment by mhdshbk** [comment]
Is this fixed yet? Because, Still I'm getting the error.

**Comment by halkar** [comment]
No, I need time to fix it (and I don't have it). @Magic73 mentioned a workaround which you can use.

**Comment by Pezzman** [comment]
PR #138 looks like it would fix this bug, can we merge the change and release a new version @halkar?

**Comment by halkar** [comment]
@Pezzman unfortunately it doesn't

**Comment by KevinAble** [comment]
@Magic73  thank you,  your work around worked for me

**Comment by DavidSilverlight** [comment]
Did you resolve the issue that was preventing the Xamarin.Forms version from working in iOS?   Based on the comments, it does not appear so.

**Comment by DavidSilverlight** [comment]
From following the comments, I am understanding that you were able to implement a workaround successfully, and all 3 platforms work for you, but that workaround did not fix it for everybody.  Is that correct?

**Comment by jamsoft** [comment]
I can't even reference this properly on iOS my refs have a yellow exclaimation mark and in code R# cannot find the RangeSlider class ... odd.

**Comment by navkar** [comment]
Resolution

1. Clone this repo. 
2. Copy all the slider*.png files into Resources directory of the IOS project. 
3. Make sure this line is present in the app delegate.cs

```csharp
[assembly: Xamarin.Forms.ExportRenderer(typeof(Xamarin.RangeSlider.Forms.RangeSlider), typeof(Xamarin.RangeSlider.Forms.RangeSliderRenderer))]
```

4. Also ensure the following is present

```csharp
var t1 = typeof(Xamarin.RangeSlider.RangeSliderControl);
var t2 = typeof(RangeSliderRenderer);
```

5. Clean and re-build the solution. 
6. Good to GO.

<img width="306" alt="screen shot 2018-11-04 at 6 36 12 pm" src="https://user-images.githubusercontent.com/7309326/47971544-a684bf80-e060-11e8-8e30-898bdcb2b4e2.png">

**Comment by aarcoraci** [comment]
thanks @navkar that worked for us!

**Comment by omersezer** [comment]
Can you create a sample project for the solution and when do you update the repository ?

**Comment by AshishRanjanPM** [comment]
Thanks

It is working for me also

**Comment by wondercrazy15** [comment]
Thanks 
It's working for me :)

**Comment by abdo-012** [comment]
Please where i can get the Png slider files
